src/router/routes.js

```diff
@@ -1,12 +1,19 @@
 const routes = [
   {
     path: "/",
-    component: () => import("layouts/MainLayout.vue"),
+    component: () => import("layouts/DetailLayout.vue"),
     children: [
       { path: "", component: () => import("pages/Index.vue") },
       { path: "/login", component: () => import("pages/auth/LoginPage.vue") },
     ],
   },
+  {
+    path: "/posts/:slug",
+    component: () => import("layouts/DetailLayout.vue"),
```

Comment:
I suggest you to put this routes inside `path: '/'` as children (nested). As we know they all consuming the same `DetailLayout`. In your approach, this will make the detail layout component re-rendered while navigating between home page and post detail page.